diffmpm/material.py

```diff
@@ -133,6 +133,75 @@ def compute_stress(self, dstrain):
         return dstrain * self.properties["E"]
 
 
+# commenting to create bingham model
+@register_pytree_node_class
+class Bingham(Material):
+    _props = (
+        "density",
+        "youngs_modulus",
+        "poisson_ratio",
+        "tau0",
+        "mu_",
+        "critical_shear_rate",
+    )
+
+    def __init__(self, material_properties):
+        self.validate_props(material_properties)
+        density = material_properties["density"]
+        youngs_modulus = material_properties["youngs_modulus"]
+        poisson_ratio = material_properties["poisson_ratio"]
+        tau0 = material_properties["tau0"]
+        mu_ = material_properties["mu_"]
+        critical_shear_rate = material_properties["critical_shear_rate"]
+        bulk_modulus = youngs_modulus / (3.0 * (1.0 - 2.0 * poisson_ratio))
+        compressibility_multiplier_=1.0
+        if(material_properties.has_key("incompressible")):
```

Comment:
Python dictionaries don't have `has_key` anymore. Is this working for you?

Reply:
I am still working it is not ready for merge, I do use black its just I have not executed the indentation command. I will check for has_key

Reply:
Yeah, no worries. I was working so I thought I'd just mention it.